bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/JsonInternalUtils.java

```diff
@@ -263,8 +263,7 @@ public static boolean isJSONObject(Object json) {
 
         MapValueImpl<BString, Object> map = new MapValueImpl<>(mapType);
         Type mapConstraint = TypeUtils.getImpliedType(mapType.getConstrainedType());
-        if (mapConstraint == null || mapConstraint.getTag() == TypeTags.ANY_TAG ||
-                mapConstraint.getTag() == TypeTags.JSON_TAG) {
+        if (mapConstraint.getTag() == TypeTags.ANY_TAG || mapConstraint.getTag() == TypeTags.JSON_TAG) {
```

Comment:
in here do we handle `mapConstraint == null`

Reply:
`TypeUtils.getImpliedType` was inferred to always return non null values by IntelliJ.

![image](https://github.com/ballerina-platform/ballerina-lang/assets/29004905/351a8d21-f553-41a3-ad43-c8729c4ab531)

See also the definition of `TypeUtils.getImpliedType`:
https://github.com/ballerina-platform/ballerina-lang/blob/c0c17f82b23240c289fabbbb9ebf5b9c894cd4aa/bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/utils/TypeUtils.java#L176-L190
It recursively calls itself until it finds a non null value or returns whatever getReferredType returns, but this has to be not null, since we access `type.getTag()`.